ddp-workspace/projects/ddp-mbc/src/app/services/languageHostRedirector.service.ts

```diff
@@ -0,0 +1,30 @@
+import { Inject, Injectable } from '@angular/core';
+import { DESTINATION_QUERY_PARAM, LANGUAGE_QUERY_PARAM } from 'ddp-sdk';
+import { MbcConfigurationService } from './mbcConfiguration.service';
+
+@Injectable({
+  providedIn: 'root',
+})
+/**
+ * Service meant to be used from app initializer to redirect from a hostname that is meant to be in a particular language,
+ * e.g., MyStudyInSpanish.org , to the "real" host, making sure after redirection that the proper language is selected
+ * Note that we leverage existing ChangeLanguageRedirectComponent
+ */
+export class LanguageHostRedirector  {
+  constructor(@Inject('ddp.config') private mbcConfig: MbcConfigurationService) {
+      const matchingHostName = mbcConfig.languageHostNames.find(each => each.hostName === window.location.hostname);
+      if (matchingHostName) {
+        this.redirectToBaseHostName(matchingHostName.languageCode);
+      }
+  }
+
+  private redirectToBaseHostName(languageCode: string): void {
+    const location = window.location;
+    const baseUrlString = `${location.protocol}//${this.mbcConfig.baseHostName}${location.port ? ':' + location.port : ''}`;
+    const redirectUrl = new URL(baseUrlString);
+    redirectUrl.pathname = '/change-language-redirect';
+    redirectUrl.searchParams.append(LANGUAGE_QUERY_PARAM, languageCode);
+    redirectUrl.searchParams.append(DESTINATION_QUERY_PARAM, '/');
+    location.href = redirectUrl.href;
+  }
+}
```

Comment:
I considered refactoring functionality to remove reliance on  ChangeLanguageRedirectComponent, but did not seem worth the hassle as this might be a unique case (this could change in future though)